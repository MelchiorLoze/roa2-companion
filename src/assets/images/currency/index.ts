import { type ImageSource } from 'expo-image';

export const CoinsIcon: ImageSource = require('./coins.png');
export const BucksIcon: ImageSource = require('./bucks.png');
export const MedalsIcon: ImageSource = require('./medals.png');
