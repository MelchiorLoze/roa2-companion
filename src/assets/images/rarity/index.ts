import { type ImageSource } from 'expo-image';

export const FrameCommonBack: ImageSource = require('./frame_common_back.png');
export const FrameRareBack: ImageSource = require('./frame_rare_back.png');
export const FrameEpicBack: ImageSource = require('./frame_epic_back.png');
export const FrameLegendaryBack: ImageSource = require('./frame_epic_back.png'); // TODO: Add legendary back frame

export const FrameCommonFront: ImageSource = require('./frame_common_front.png');
export const FrameRareFront: ImageSource = require('./frame_rare_front.png');
export const FrameEpicFront: ImageSource = require('./frame_epic_front.png');
export const FrameLegendaryFront: ImageSource = require('./frame_epic_front.png'); // TODO: Add legendary front frame
