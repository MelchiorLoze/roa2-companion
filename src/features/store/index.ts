// COMPONENTS
export { ItemList } from './components/ItemList/ItemList';
export { TimeCountdown } from './components/TimeCountdown/TimeCountdown';

// HOOKS
export { useRotatingCoinShop } from './hooks/business/useRotatingCoinShop/useRotatingCoinShop';
export { usePurchaseInventoryItems } from './hooks/data/usePurchaseInventoryItems/usePurchaseInventoryItems';

// TYPES
export type { Item } from './types/item';
