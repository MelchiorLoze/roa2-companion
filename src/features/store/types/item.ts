import type { DateTime } from 'luxon';

import type { CurrencyId } from '@/types/currency';

import type { Rarity, RarityValue } from './rarity';

export const enum Category {
  DEATHEFFECT = 'deatheffect',
  EMOTE = 'emote',
  ICON = 'icon',
  PALETTE = 'palette',
  PLATFORM = 'platform',
  SKIN = 'skin',
  STAGESKIN = 'stageskin',
  TAUNT = 'taunt',
}

export const CATEGORY_LABELS: Readonly<Record<Category, string>> = Object.freeze({
  [Category.DEATHEFFECT]: 'DEATH',
  [Category.EMOTE]: 'EMOTE',
  [Category.ICON]: 'ICON',
  [Category.PALETTE]: 'PALETTE',
  [Category.PLATFORM]: 'PLATFORM',
  [Category.SKIN]: 'SKIN',
  [Category.STAGESKIN]: 'STAGE SKIN',
  [Category.TAUNT]: 'TAUNT',
});

export type ItemDto = {
  Id: string;
  Title: { NEUTRAL: string };
  ContentType: Category;
  PriceOptions?: {
    Prices: {
      Amounts: {
        ItemId: CurrencyId;
        Amount: number;
      }[];
    }[];
  };
  DisplayProperties: {
    rarity: RarityValue;
  };
};

export type Item = {
  id: string;
  title: string;
  category: Category;
  rarity: Rarity;
  coinPrice?: number;
  buckPrice?: number;
};

export type ItemWithCoinPrice = Item & Required<Pick<Item, 'coinPrice'>>;

export type RotationalCoinStore = {
  expirationDate: DateTime;
  itemIds: Item['id'][];
};
