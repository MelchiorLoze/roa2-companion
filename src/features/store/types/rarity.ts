import type { ImageSource } from 'expo-image';

import { CommonIcon, EpicIcon, LegendaryIcon, RareIcon } from '../assets';

export const enum RarityValue {
  COMMON = 1,
  RARE = 2,
  EPIC = 3,
  LEGENDARY = 4,
}

export const enum Rarity {
  COMMON = 'common',
  RARE = 'rare',
  EPIC = 'epic',
  LEGENDARY = 'legendary',
}

export const RARITY_VALUES_MAPPING: Readonly<Record<RarityValue, Rarity>> = Object.freeze({
  [RarityValue.COMMON]: Rarity.COMMON,
  [RarityValue.RARE]: Rarity.RARE,
  [RarityValue.EPIC]: Rarity.EPIC,
  [RarityValue.LEGENDARY]: Rarity.LEGENDARY,
});

export const RARITY_ICONS: Readonly<Record<Rarity, ImageSource>> = Object.freeze({
  [Rarity.COMMON]: CommonIcon,
  [Rarity.RARE]: RareIcon,
  [Rarity.EPIC]: EpicIcon,
  [Rarity.LEGENDARY]: LegendaryIcon,
});
