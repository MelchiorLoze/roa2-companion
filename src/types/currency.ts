import type { ImageSource } from 'expo-image';

import { BucksIcon, CoinsIcon, MedalsIcon } from '@/assets/images';

export const enum CurrencyId {
  COINS = 'b3f4a8f0-dd58-4e3f-ae0a-7a17418fc903',
  BUCKS = 'ed4812be-4dcd-446b-b61e-96d8be8f6121',
  MEDALS = 'daf33d00-75b5-4b0e-8cbc-8219b9dcb7ca',
}

export const enum Currency {
  COINS = 'coins',
  BUCKS = 'bucks',
  MEDALS = 'medals',
}

export const CURRENCY_ICONS: Readonly<Record<Currency, ImageSource>> = Object.freeze({
  [Currency.COINS]: CoinsIcon,
  [Currency.BUCKS]: BucksIcon,
  [Currency.MEDALS]: MedalsIcon,
});
