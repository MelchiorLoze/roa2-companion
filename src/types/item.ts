import type { ImageSource } from 'expo-image';

import { CommonIcon, EpicIcon, LegendaryIcon, RareIcon } from '@/assets/images/rarity';

import type { CurrencyId } from './currency';

export const enum Category {
  DEATHEFFECT = 'deatheffect',
  EMOTE = 'emote',
  ICON = 'icon',
  PALETTE = 'palette',
  PLATFORM = 'platform',
  SKIN = 'skin',
  STAGESKIN = 'stageskin',
  TAUNT = 'taunt',
}

export const CATEGORY_LABELS: Readonly<Record<Category, string>> = Object.freeze({
  [Category.DEATHEFFECT]: 'DEATH',
  [Category.EMOTE]: 'EMOTE',
  [Category.ICON]: 'ICON',
  [Category.PALETTE]: 'PALETTE',
  [Category.PLATFORM]: 'PLATFORM',
  [Category.SKIN]: 'SKIN',
  [Category.STAGESKIN]: 'STAGE SKIN',
  [Category.TAUNT]: 'TAUNT',
});

export const enum RarityValue {
  COMMON = 1,
  RARE = 2,
  EPIC = 3,
  LEGENDARY = 4,
}

export const enum Rarity {
  COMMON = 'common',
  RARE = 'rare',
  EPIC = 'epic',
  LEGENDARY = 'legendary',
}

export const RARITY_VALUES_MAPPING: Readonly<Record<RarityValue, Rarity>> = Object.freeze({
  [RarityValue.COMMON]: Rarity.COMMON,
  [RarityValue.RARE]: Rarity.RARE,
  [RarityValue.EPIC]: Rarity.EPIC,
  [RarityValue.LEGENDARY]: Rarity.LEGENDARY,
});

export const RARITY_ICONS: Readonly<Record<Rarity, ImageSource>> = Object.freeze({
  [Rarity.COMMON]: CommonIcon,
  [Rarity.RARE]: RareIcon,
  [Rarity.EPIC]: EpicIcon,
  [Rarity.LEGENDARY]: LegendaryIcon,
});

export type ItemDto = {
  Id: string;
  Title: { NEUTRAL: string };
  ContentType: Category;
  PriceOptions?: {
    Prices: {
      Amounts: {
        ItemId: CurrencyId;
        Amount: number;
      }[];
    }[];
  };
  DisplayProperties: {
    rarity: RarityValue;
  };
};

export type Item = {
  id: string;
  name: string;
  category: Category;
  rarity: Rarity;
  coinPrice?: number;
  buckPrice?: number;
};
