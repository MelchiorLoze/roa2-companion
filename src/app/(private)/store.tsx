import { LinearGradient } from 'expo-linear-gradient';
import { useFocusEffect } from 'expo-router';
import { useCallback, useState } from 'react';
import { Text, View } from 'react-native';
import { StyleSheet, useUnistyles } from 'react-native-unistyles';

import { Button, Dialog, Spinner } from '@/components';
import { type Item, ItemList, TimeCountdown, usePurchaseInventoryItems, useRotatingCoinShop } from '@/features/store';
import { CATEGORY_LABELS } from '@/features/store/types/item';
import { CurrencyId } from '@/types/currency';

export default function Store() {
  const { theme } = useUnistyles();

  const [selectedItem, setSelectedItem] = useState<Item | null>(null);
  const { items, expirationDate, isLoading: isCoinShopLoading } = useRotatingCoinShop();
  const { purchase, isLoading: isPurchaseLoading } = usePurchaseInventoryItems();

  const handlePurchase = () => {
    if (!selectedItem?.coinPrice) return;
    purchase({
      id: selectedItem.id,
      price: { value: selectedItem.coinPrice, currencyId: CurrencyId.COINS },
    });
    closeDialog();
  };

  const openDialog = (item: Item) => {
    if (item.coinPrice) setSelectedItem(item);
  };

  const closeDialog = () => setSelectedItem(null);

  useFocusEffect(useCallback(closeDialog, []));

  if (isCoinShopLoading || isPurchaseLoading) return <Spinner />;

  return (
    <>
      <View style={styles.container}>
        {expirationDate && (
          <LinearGradient colors={theme.color.labelGradient} end={[1, 0]} start={[0, 0]} style={styles.titleContainer}>
            <Text style={styles.title}>Items refresh in:</Text>
            <TimeCountdown date={expirationDate} style={styles.title} />
          </LinearGradient>
        )}
        <ItemList items={items} onSelect={openDialog} />
      </View>
      {selectedItem && (
        <Dialog onClose={closeDialog}>
          <Text style={styles.title}>
            Are you sure you want to buy the {CATEGORY_LABELS[selectedItem.category]} {selectedItem.title} for{' '}
            {selectedItem.coinPrice}?
          </Text>
          <View style={styles.buttonContainer}>
            <Button label="Yes" onPress={handlePurchase} />
            <Button label="No" onPress={closeDialog} />
          </View>
        </Dialog>
      )}
    </>
  );
}

const styles = StyleSheet.create((theme) => ({
  container: {
    flex: 1,
    padding: theme.spacing.l,
    gap: theme.spacing.l,
  },
  titleContainer: {
    flexDirection: 'row',
    justifyContent: 'flex-end',
    alignItems: 'center',
    width: '100%',
    padding: theme.spacing.s,
    paddingRight: theme.spacing.m,
    gap: theme.spacing.xs,
  },
  title: {
    fontFamily: theme.font.secondary.bold,
    fontSize: 18,
    color: theme.color.white,
    textTransform: 'uppercase',
  },
  buttonContainer: {
    flexDirection: 'row',
    justifyContent: 'space-around',
  },
}));
