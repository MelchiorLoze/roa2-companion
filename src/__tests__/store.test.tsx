import { fireEvent, render, screen, within } from '@testing-library/react-native';
import { DateTime } from 'luxon';

import Store from '@/app/(private)/store';
import { type Item, usePurchaseInventoryItems, useRotatingCoinShop } from '@/features/store';
import { Category, CATEGORY_LABELS } from '@/features/store/types/item';
import { Rarity } from '@/features/store/types/rarity';
import { CurrencyId } from '@/types/currency';

jest.mock('expo-router');

jest.mock<typeof import('@/features/store')>('@/features/store', () => ({
  ...jest.requireActual('@/features/store'),
  useRotatingCoinShop: jest.fn(),
  usePurchaseInventoryItems: jest.fn(),
}));
const useRotatingCoinShopMock = jest.mocked(useRotatingCoinShop);
const usePurchaseInventoryItemsMock = jest.mocked(usePurchaseInventoryItems);
const purchaseMock = jest.fn();

const items: Item[] = [
  {
    id: '1',
    title: 'Item 1',
    rarity: Rarity.COMMON,
    category: Category.ICON,
    coinPrice: 2000,
    buckPrice: 20,
  },
  {
    id: '2',
    title: 'Item 2',
    rarity: Rarity.EPIC,
    category: Category.DEATHEFFECT,
    buckPrice: 500,
  },
];

const confirmationDialogTitle = (item: Item) =>
  `Are you sure you want to buy the ${CATEGORY_LABELS[item.category]} ${item.title} for ${item.coinPrice}?`;

const renderComponent = () => {
  const result = render(<Store />);

  expect(useRotatingCoinShopMock).toHaveBeenCalledTimes(1);
  expect(usePurchaseInventoryItemsMock).toHaveBeenCalledTimes(1);
  expect(purchaseMock).not.toHaveBeenCalled();

  return result;
};

describe('Store', () => {
  beforeEach(() => {
    useRotatingCoinShopMock.mockReturnValue({
      items,
      expirationDate: DateTime.utc().plus({ day: 1 }),
      isLoading: false,
    });
    usePurchaseInventoryItemsMock.mockReturnValue({
      purchase: purchaseMock,
      isLoading: false,
      isError: false,
    });
  });

  it('matches the snapshot', () => {
    const tree = renderComponent().toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders the items of the store', () => {
    renderComponent();

    screen.getByText('Items refresh in:');
    const itemCards = screen.getAllByRole('button');
    expect(itemCards).toHaveLength(items.length);
    within(itemCards[0]).getByText(items[0].title);
    within(itemCards[1]).getByText(items[1].title);

    fireEvent.press(itemCards[0]);
    const withinDialog = within(screen.getByTestId('dialog'));

    withinDialog.getByText(confirmationDialogTitle(items[0]));
    withinDialog.getByRole('button', { name: 'Yes' });
    withinDialog.getByRole('button', { name: 'No' });
  });

  it('does not show the confirmation dialog when the selected item does not have a coin price', () => {
    expect(items[1].coinPrice).toBeUndefined();

    renderComponent();

    const itemCards = screen.getAllByRole('button');
    within(itemCards[1]).getByText(items[1].title);

    fireEvent.press(itemCards[1]);

    expect(screen.queryByText(confirmationDialogTitle(items[1]))).toBeNull();
  });

  it('closes the confirmation dialog when the overlay is pressed', () => {
    renderComponent();

    const itemCards = screen.getAllByRole('button');
    fireEvent.press(itemCards[0]);
    const withinDialog = within(screen.getByTestId('dialog'));

    withinDialog.getByText(confirmationDialogTitle(items[0]));

    const overlay = screen.getByTestId('overlay');
    fireEvent.press(overlay);

    expect(screen.queryByText(confirmationDialogTitle(items[0]))).toBeNull();
  });

  it('closes the confirmation dialog when "No" is pressed', () => {
    renderComponent();

    const itemCards = screen.getAllByRole('button');
    fireEvent.press(itemCards[0]);
    const withinDialog = within(screen.getByTestId('dialog'));

    withinDialog.getByText(confirmationDialogTitle(items[0]));

    const noButton = withinDialog.getByRole('button', { name: 'No' });
    fireEvent.press(noButton);

    expect(screen.queryByText(confirmationDialogTitle(items[0]))).toBeNull();
  });

  it('calls purchase when "Yes" is pressed', () => {
    renderComponent();

    const itemCards = screen.getAllByRole('button');
    fireEvent.press(itemCards[0]);
    const withinDialog = within(screen.getByTestId('dialog'));

    withinDialog.getByText(confirmationDialogTitle(items[0]));

    const yesButton = withinDialog.getByRole('button', { name: 'Yes' });
    fireEvent.press(yesButton);

    expect(purchaseMock).toHaveBeenCalledTimes(1);
    expect(purchaseMock).toHaveBeenCalledWith({
      id: items[0].id,
      price: { value: items[0].coinPrice, currencyId: CurrencyId.COINS },
    });
    expect(screen.queryByText(confirmationDialogTitle(items[0]))).toBeNull();
  });

  it('displays a spinner when the store items are loading', () => {
    useRotatingCoinShopMock.mockReturnValue({
      items: [],
      expirationDate: undefined,
      isLoading: true,
    });

    renderComponent();

    screen.getByAccessibilityHint('Loading...');
  });

  it('displays a spinner when the purchase is loading', () => {
    usePurchaseInventoryItemsMock.mockReturnValue({
      purchase: purchaseMock,
      isLoading: true,
      isError: false,
    });

    renderComponent();

    screen.getByAccessibilityHint('Loading...');
  });
});
