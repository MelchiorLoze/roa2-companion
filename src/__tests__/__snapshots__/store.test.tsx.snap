// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Store matches the snapshot 1`] = `
<View
  style={
    {
      "flex": 1,
      "gap": 16,
      "padding": 16,
    }
  }
>
  <ViewManagerAdapter_ExpoLinearGradient
    colors={
      [
        0,
        4278190080,
      ]
    }
    endPoint={
      [
        1,
        0,
      ]
    }
    startPoint={
      [
        0,
        0,
      ]
    }
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "gap": 4,
        "justifyContent": "flex-end",
        "padding": 8,
        "paddingRight": 12,
        "width": "100%",
      }
    }
  >
    <Text
      style={
        {
          "color": "white",
          "fontFamily": "AgencyFB-Bold",
          "fontSize": 18,
          "textTransform": "uppercase",
        }
      }
    >
      Items refresh in:
    </Text>
    <View>
      <Text
        style={
          [
            {
              "color": "white",
              "fontFamily": "AgencyFB-Bold",
              "fontSize": 18,
              "textTransform": "uppercase",
            },
            {
              "position": "absolute",
              "right": 0,
            },
          ]
        }
      >
        23
        h 
        59
        m 
        59
        s
      </Text>
      <Text
        style={
          [
            {
              "color": "white",
              "fontFamily": "AgencyFB-Bold",
              "fontSize": 18,
              "textTransform": "uppercase",
            },
            {
              "color": "transparent",
            },
          ]
        }
      >
        23
        h 
        59
        m 00s
      </Text>
    </View>
  </ViewManagerAdapter_ExpoLinearGradient>
  <RCTScrollView
    contentContainerStyle={
      {
        "gap": 16,
      }
    }
    data={
      [
        {
          "buckPrice": 20,
          "category": "icon",
          "coinPrice": 2000,
          "id": "1",
          "rarity": "common",
          "title": "Item 1",
        },
        {
          "buckPrice": 500,
          "category": "deatheffect",
          "id": "2",
          "rarity": "epic",
          "title": "Item 2",
        },
      ]
    }
    getItem={[Function]}
    getItemCount={[Function]}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={0.0001}
    stickyHeaderIndices={[]}
    viewabilityConfigCallbackPairs={[]}
  >
    <View>
      <View
        onFocusCapture={[Function]}
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            [
              {
                "flexDirection": "row",
              },
              {
                "gap": 16,
              },
            ]
          }
        >
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            role="button"
            style={
              {
                "flex": 0.5,
              }
            }
          >
            <ViewManagerAdapter_ExpoLinearGradient
              colors={
                [
                  4291481343,
                  4285232860,
                ]
              }
              style={
                {
                  "flex": 1,
                  "padding": 2,
                }
              }
            >
              <ViewManagerAdapter_ExpoLinearGradient
                colors={
                  [
                    4279636041,
                    4279043869,
                  ]
                }
                style={
                  {
                    "flex": 1,
                    "gap": 16,
                    "justifyContent": "space-between",
                    "padding": 8,
                  }
                }
              >
                <View
                  style={
                    {
                      "flexDirection": "row",
                      "gap": 4,
                      "justifyContent": "space-between",
                    }
                  }
                >
                  <Text
                    style={
                      [
                        {
                          "color": "white",
                          "flex": 1,
                          "fontFamily": "AgencyFB-Bold",
                          "fontSize": 16,
                          "textTransform": "uppercase",
                        },
                        false,
                      ]
                    }
                  >
                    Item 1
                  </Text>
                  <ViewManagerAdapter_ExpoImage
                    containerViewRef={"[React.ref]"}
                    contentFit="contain"
                    contentPosition={
                      {
                        "left": "50%",
                        "top": "50%",
                      }
                    }
                    height={16}
                    marginTop={2}
                    nativeViewRef={"[React.ref]"}
                    onError={[Function]}
                    onLoad={[Function]}
                    onLoadStart={[Function]}
                    onProgress={[Function]}
                    placeholder={[]}
                    source={
                      [
                        {
                          "testUri": "../../../features/store/assets/rarity/common.webp",
                        },
                      ]
                    }
                    style={
                      {
                        "height": 16,
                        "marginTop": 2,
                        "width": 16,
                      }
                    }
                    transition={null}
                    width={16}
                  />
                </View>
                <View
                  style={
                    {
                      "flexDirection": "row",
                      "justifyContent": "space-between",
                    }
                  }
                >
                  <View
                    style={
                      {
                        "alignItems": "center",
                        "flexDirection": "row",
                        "gap": 4,
                      }
                    }
                  >
                    <ViewManagerAdapter_ExpoImage
                      containerViewRef={"[React.ref]"}
                      contentFit="contain"
                      contentPosition={
                        {
                          "left": "50%",
                          "top": "50%",
                        }
                      }
                      height={20}
                      nativeViewRef={"[React.ref]"}
                      onError={[Function]}
                      onLoad={[Function]}
                      onLoadStart={[Function]}
                      onProgress={[Function]}
                      placeholder={[]}
                      source={
                        [
                          {
                            "testUri": "../../../assets/images/currency/coins.png",
                          },
                        ]
                      }
                      style={
                        {
                          "height": 20,
                          "width": 20,
                        }
                      }
                      transition={null}
                      width={20}
                    />
                    <Text
                      style={
                        [
                          {
                            "color": "white",
                            "fontFamily": "AgencyFB-Bold",
                            "fontSize": 14,
                            "textTransform": "uppercase",
                          },
                          false,
                        ]
                      }
                    >
                      2000
                    </Text>
                  </View>
                </View>
              </ViewManagerAdapter_ExpoLinearGradient>
            </ViewManagerAdapter_ExpoLinearGradient>
          </View>
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            role="button"
            style={
              {
                "flex": 0.5,
              }
            }
          >
            <ViewManagerAdapter_ExpoLinearGradient
              colors={
                [
                  4291481343,
                  4285232860,
                ]
              }
              style={
                {
                  "flex": 1,
                  "padding": 2,
                }
              }
            >
              <ViewManagerAdapter_ExpoLinearGradient
                colors={
                  [
                    4279636041,
                    4279043869,
                  ]
                }
                style={
                  {
                    "flex": 1,
                    "gap": 16,
                    "justifyContent": "space-between",
                    "padding": 8,
                  }
                }
              >
                <View
                  style={
                    {
                      "flexDirection": "row",
                      "gap": 4,
                      "justifyContent": "space-between",
                    }
                  }
                >
                  <Text
                    style={
                      [
                        {
                          "color": "white",
                          "flex": 1,
                          "fontFamily": "AgencyFB-Bold",
                          "fontSize": 16,
                          "textTransform": "uppercase",
                        },
                        false,
                      ]
                    }
                  >
                    Item 2
                  </Text>
                  <ViewManagerAdapter_ExpoImage
                    containerViewRef={"[React.ref]"}
                    contentFit="contain"
                    contentPosition={
                      {
                        "left": "50%",
                        "top": "50%",
                      }
                    }
                    height={16}
                    marginTop={2}
                    nativeViewRef={"[React.ref]"}
                    onError={[Function]}
                    onLoad={[Function]}
                    onLoadStart={[Function]}
                    onProgress={[Function]}
                    placeholder={[]}
                    source={
                      [
                        {
                          "testUri": "../../../features/store/assets/rarity/epic.webp",
                        },
                      ]
                    }
                    style={
                      {
                        "height": 16,
                        "marginTop": 2,
                        "width": 16,
                      }
                    }
                    transition={null}
                    width={16}
                  />
                </View>
                <View
                  style={
                    {
                      "flexDirection": "row",
                      "justifyContent": "space-between",
                    }
                  }
                />
              </ViewManagerAdapter_ExpoLinearGradient>
            </ViewManagerAdapter_ExpoLinearGradient>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
